src/layouts/BaseLayout/BaseLayout.jsx

```diff
@@ -25,24 +26,34 @@ const FullPageLoader = () => (
 )
 
 const OnboardingOrChildren = ({ children }) => {
-  const { isFullExperience, showAgreeToTerms, showDefaultOrgSelector } =
-    useUserAccessGate()
+  const {
+    isFullExperience,
+    showAgreeToTerms,
+    showDefaultOrgSelector,
+    redirectToSyncPage,
+  } = useUserAccessGate()
 
   const { isImpersonating } = useImpersonate()
 
-  if (showAgreeToTerms && !isFullExperience)
+  if (showAgreeToTerms && !isFullExperience) {
     return (
       <Suspense fallback={null}>
         <TermsOfService />
       </Suspense>
     )
+  }
+
+  if (redirectToSyncPage && !isFullExperience) {
+    return <Redirect to="/sync" />
```

Comment:
what if i'm already in /sync route? does this mean i'll still get redirected to /sync?

Reply:
This code block here checks to see if you're already on the `/sync` route, and if you are it will set `redirectToSyncPage` to `false`.

```jsx
const onSyncPage = currentRoute.path === '/sync'
if (sentryLoginProvider && !isGuest && !onSyncPage) {
  // owners array contains a list of the synced providers
  // if it is zero then they haven't synced any other providers
  redirectToSyncPage = isEqual(internalUser?.owners?.length, 0)
}
```

Reply:
makes sense